src/badges/views.py

```diff
@@ -106,12 +105,22 @@ def badge_create(request):
 @non_public_only_view
 @staff_member_required
 def badge_copy(request, badge_id):
-    new_badge = get_object_or_404(Badge, pk=badge_id)
-    new_badge.pk = None  # autogen a new primary key (quest_id by default)
-    new_badge.import_id = uuid.uuid4()
-    new_badge.name = "Copy of " + new_badge.name
+    badge = get_object_or_404(Badge, pk=badge_id)
+    
+    form_data_initial = model_to_dict(badge)
```

Comment:
I think you should leave the old code.  Why change to this?  Also, you are now copying the uuid I think, which needs to be unique (and if your code passes tests, then there is a missing test that needs to ensure a copy has a unique ID!)

Note if the docs say this is the prefered way to doing it, link to docs and we can do it this way, otherwise, for consistancy throughout the project, setting the id to None is the way the rest of the project copies model objects.

Reply:
i dont think theres a need to copy the model if we dont need to do it. Old code also wont work by adding tags to instance. 

using ``initial`` over ``instance`` seems to be the proper way to populate the form fields anyway. As there really isn't a mention of ``instance`` arg in the docs
https://docs.djangoproject.com/en/4.1/ref/forms/api/

Reply:
also uuid does not matter since im not using an instance of Quest, just passing args to initial data

Reply:
Why won't the old code work?  You just need to make sure you save the model object first (so it gets an ID and is added to the DB), then you can add the tags.

Reply:
I don't like this `form_data_initial = model_to_dict(badge)` because it's implicit.  I like explicit.  Zen of Python!

Reply:
if i save the model obj wont that defeat the purpose of QuestCopy being apart of CreateView?

what if i want to back out of quest copy creation, then the model will still be in the db.

if there is already an instance of ``Quest - Copy`` and i copy ``Quest`` then it will give me a ``404`` because ``Quest - Copy`` already exists.

Reply:
also this doesnt work
```
new_quest.tags = copied_quest.tags
kwargs['instance'] = new_quest
```